### ISSUE TITLE
Stuck in Hypothesis generation

### ISSUE THREAD

**Issue by max123465** [cvlab-dresden/DSAC#18]
When I doing the # end-to-end training using DSAC
It will get stuck in Sampling 256 hypotheses and generate no error.
-------
here is the cmd 

Loading test set ...
Reading file names... 
Loading script: train_score.lua
TORCH: Loading network from file: score_model_init.net
Loading script: train_obj.lua
TORCH: Loading network from file: obj_model_init.net
Processing test image 0 of 2000.
Predicting object coordinates.
TORCH: Doing a forward pass for 1600 patches.
CNN prediction took 0.671922s.
Sampling 256 hypotheses.

**Comment by max123465** [comment]
I found that the problem is in cnn.h, estObj always return 0 array.
I had check my env, with lua 5.3, torch 7, luajut 2.1, opencv 2.4.13.
I have no idea where the problem is... 
Could anyone help me  : (

**Comment by max123465** [comment]
I can do 
pre-train scene coordinate regression CNN
and pre-train score regression CNN
But when i want to do # end-to-end training using DSAC,
it stuck at searching hypothesis.
So now i think the problem is related to "safeSolvepnp" here.
Dose anyone get the same problem?

**Comment by ebrach** [comment]
Infinite hypotheses sampling is often caused by a faulty LUA setup. (Unfortunately, the code has poor error handling. Any errors on the LUA side will be swallowed by C++)

See https://github.com/cvlab-dresden/DSAC/issues/17
